Classes/Widgets/Provider/PagesWithoutDescriptionDataProvider.php

```diff
@@ -41,29 +41,24 @@ public function getPages(): array
         ];
 
         $items = [];
-        $counter = 0;
-        $iterator = 0;
 
-        while ($counter < $this->limit) {
-            $row = $queryBuilder
-                ->select('*')
-                ->from('pages')
-                ->where(...$constraints)
-                ->orderBy('tstamp', 'DESC')
-                ->setFirstResult($iterator)
-                ->setMaxResults(1)
-                ->execute()
-                ->fetch();
-
-            $iterator++;
+        $results = $queryBuilder
```

Comment:
For an administrator this is a good way to do so, but if you are an editor and only allowed to see a specific part of the tree, this might end up showing no pages at all as the first items might be all pages you don't have access to. So I think we should solve this problem a little bit different. Maybe we should set the limit to the max of the number of rows in total.

Reply:
@haassie 
Can you watch this again?
I reverted the old code and added an if-clause if the $row is false. 
That worked for me.